doozer/doozerlib/cli/images_streams.py

```
@@ -394,14 +401,40 @@ def _get_upstreaming_entries(runtime, stream_names=None):
                     dict_to_model=image_meta.config.content.source.ci_alignment.primitive()
                 )  # Make a copy
-                upstream_entry['image'] = (
-                    image_meta.pull_url()
-                )  # Make the image metadata entry match what would exist in streams.yml.
+
+                # Determine which build system to use for getting the image pullspec
+                if runtime.build_system == 'konflux':
+                    # Get the pullspec from Konflux DB
+                    async def get_konflux_pullspec():
+                        build = await runtime.konflux_db.get_latest_build(
+                            name=image_meta.distgit_key,
```

Ruff (0.14.3)

Function definition does not bind loop variable `image_meta`

(B023)